### PR TITLE
Clean up of docs/user_guide.md #81

### LINKED ISSUE #81: Clean up of docs/user_guide.md
The file docs/user_guide.md is the landing page for the doxygen generated documentation. It contains a high-level description of the library.

The current page is the first pass of converting some legacy documentation, and it needs some more work.

Most importantly, all functions/subroutines need to have "()" added after their name, so that they get hyperlined to the proper subprogram documentation. For example:

`To start a new GRIB2 message, call subroutine GRIBCREATE. GRIBCREATE encodes Sections 0 and 1 at the beginning of the message. This routine must be used to create each message.
`

Should become:

`To start a new GRIB2 message, call subroutine GRIBCREATE(). GRIBCREATE() encodes Sections 0 and 1 at the beginning of the message. This routine must be used to create each message.
`

This will cause the subprogram name GRIBCREATE to automatically be linked, so that the user can click on it and jump to the documentation for GRIBCREATE. This needs to happen with all subprogram names in the document.

Further down in the user_guide.md, there is some example code which is not being formatted correctly in doxygen, and that needs to be fixed.

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
issue #81 
This has been add and tested on
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/files.html

### CODE REVIEW

File: docs/user_guide.md [hunk 1]
```diff
@@ -36,72 +36,72 @@ a number of different grids, several routines are used to encode a message.
...
 within one or more GRIB2 messages.
 
-To start a new GRIB2 message, call subroutine GRIBCREATE.  GRIBCREATE 
-encodes Sections 0 and 1 at the beginning of the message.  This routine 
+To start a new GRIB2 message, call subroutine GRIBCREATE(). GRIBCREATE()
+encodes Sections 0 and 1 at the beginning of the message. This routine 
 must be used to create each message.
 
-Subroutine ADDLOCAL can be used to add a Local Use Section ( Section 2 ).
+Subroutine ADDLOCAL() can be used to add a Local Use Section (Section 2).
```
Comment (GeorgeGayno-NOAA) [inline docs/user_guide.md]: I see you added () to the subroutine names. However, doxygen did not create links to these routines.

**Review by edwardhartnett — APPROVED** [review]

File: docs/user_guide.md (hunk 1 above)
Comment (Hang-Lei-NOAA) [reply docs/user_guide.md]: Do you know how to set it?
I add *.mod to the files, but it still not treat this file as code.

Comment (GeorgeGayno-NOAA) [reply docs/user_guide.md]: I was looking at other repos for an example, but did not find one. Maybe it is not possible to create links to source code.

Comment (edwardhartnett) [reply docs/user_guide.md]: Try lower-casing the function name...

**Comment by Hang-Lei-NOAA** [comment]
Okay, thanks both.

File: docs/user_guide.md (hunk 1 above)
Comment (Hang-Lei-NOAA) [reply docs/user_guide.md]: Lower-casing works.

**Pull request merged**